llvm/lib/Target/RISCV/RISCVAsmPrinter.cpp

```diff
@@ -132,6 +128,24 @@ class RISCVAsmPrinter : public AsmPrinter {
     /// The size in bytes of the stack frame, 0 if not used.
     uint32_t stackSize = 0;
   };
+
+  /**
+   * Struct describing sealing key-like compartment export objects.
+   */
+  struct SealingKeyCompartmentExport {
+    /// The name of the sealing key type.
+    std::string SealingKeyTypeName;
+  };
+
+  /**
+   * Struct describing compartment exports that must be emitted for this
+   * compilation unit.
+   */
+  struct CompartmentExport {
+    /// The compartment name for the object.
+    std::string CompartmentName;
+    std::variant<FunctionCompartmentExport, SealingKeyCompartmentExport> Object;
```

Comment:
Alternatively, you could just have two vectors, one for `FunctionCompartmentExport`'s and one for `SealingKeyCompartmentExport`'s ;-)